ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/Invoker.java

```diff
@@ -53,16 +69,36 @@
  * @since 2.0.0
  */
 public abstract class Invoker extends DiagnosticReporter {
+    public static final String MODULE_NOT_FOUND_CODE = "BCE2003";
+    // TODO: After configurables can be supported, change this to that file location
+    private static final Path CONFIG_PATH = Paths.get(System.getProperty("user.dir"), "Config.toml");
+    /* Constants related to execution */
+    private static final String MODULE_INIT_CLASS_NAME = "$_init";
+    private static final String CONFIGURE_INIT_CLASS_NAME = "$ConfigurationMapper";
+    private static final String MODULE_INIT_METHOD_NAME = "$moduleInit";
+    private static final String CONFIGURE_INIT_METHOD_NAME = "$configureInit";
+    private static final String MODULE_MAIN_METHOD_NAME = "main";
+    /* Constants related to temp files */
+    private static final String TEMP_FILE_PREFIX = "main-";
+    private static final String TEMP_FILE_SUFFIX = ".bal";
     private static final String NON_ACCESSIBLE_TYPE_CODE = "BCE2037";
-    private static final boolean USE_TEMP_FILE = true;
 
+    /**
+     * Scheduler used to run the init and main methods
+     * of the generated classes.
+     */
+    private final Scheduler scheduler;
     /**
      * File object that is used to create projects and write.
      * Depending on USE_TEMP_FILE flag, this may be either a file in cwd
      * or a temp file.
      */
     private File bufferFile;
 
+    protected Invoker() {
+        this.scheduler = new Scheduler(false);
```

Comment:
It is incorrect to new a scheduler. This class is exported to only few packages and not supposed to be used by all.

Reply:
The issue is that shell executes each snippet by compiling and running a function on the generated jar. To invoke them with reflection, we need a `Schedular`. And shell invocations are not one program actually, each snippet runs as its own program, so during a shell session it will run multiple programs.

Reply:
same as above